### PR TITLE
Enable npm install from source

### LINKED ISSUE #2237: Fix npm install from source
Node bindings are now included in this repo but it is impossible to install mapbox-gl-native from source using npm because

1. building mapbox-gl-native from source relies on the ability to install git submodules
2. npm strips out all git info (including submodules)

In order to make it possible to install from source, we will

 - [x] require that mason be installed globally (or clone the latest release?)
 - [x] ~~install~~ build geojsonvt via mason
 - [x] modify build scripts to ignore irrelevant missing dependencies
 - [x] fix [iOS release](https://travis-ci.org/mapbox/mapbox-gl-native/jobs/78480683) and [linux release](https://travis-ci.org/mapbox/mapbox-gl-native/jobs/78480679) builds

ref #1522 

cc @jfirebaugh @mikemorris @tmcw @incanus @bsudekum

### PR DESCRIPTION
**Pull request by lucaswoj**
cc @jfirebaugh @mikemorris @incanus @1ec5 

resolves #2237

### CODE REVIEW

File: configure [hunk 1]
```diff
@@ -15,8 +15,15 @@ function abort { >&2 echo -e "\033[1m\033[31m$1\033[0m"; exit 1; }
...
-# Install mason
-export PATH="`pwd`/.mason:${PATH}" MASON_DIR="`pwd`/.mason"
+if [ -d "`pwd`/.git" ]; then
+    info "This build is within a git repository"
+    export MASON_DIR="`pwd`/.mason"
+    export PATH="${MASON_DIR}:${PATH}"
+else
+    info "This build is NOT within a git repository"
+    which mason || abort "You must install mason to build mapbox-gl-native (https://github.com/mapbox/mason)"
+    export MASON_DIR="$(dirname $(readlink $(which mason)))"
```
Comment (lucaswoj) [inline configure]: Is there a better way to do this?

File: scripts/build-version.py [hunk 2]
```diff
@@ -1,22 +1,23 @@
...
-import sys, os, errno, re
+import sys, os, errno, re, subprocess
 
+DEFAULT_TAG = [0, 0, 0]
+DEFAULT_REV = 'unknown'
 
-output_dir = sys.argv[1]
+def is_git_repo():
+    try:
+        subprocess.check_output("which git", shell=True)
```
Comment (lucaswoj) [inline scripts/build-version.py]: This line is unnecessary

File: scripts/main.mk [hunk 3]
```diff
@@ -16,13 +16,16 @@ HOST_VERSION ?= $(BUILD_VERSION)
...
+	CONFIGURE_FILES += scripts/$(HOST)/$(HOST_VERSION)/configure.sh
+endif
+
 HOST_SLUG = $(HOST)-$(HOST_VERSION)
 CONFIGURE_FILES = scripts/$(HOST)/configure.sh
 ifneq (,$(wildcard scripts/$(HOST)/$(HOST_VERSION)/configure.sh))
 	CONFIGURE_FILES += scripts/$(HOST)/$(HOST_VERSION)/configure.sh
 endif
 
-
```
Comment (lucaswoj) [inline scripts/main.mk]: Revert whitespace change

File: gyp/version.gypi [hunk 4]
```diff
@@ -13,7 +13,7 @@
           'outputs': [
             '<(SHARED_INTERMEDIATE_DIR)/include/mbgl/util/version.hpp',
           ],
-          'action': ['<@(_inputs)', '<(SHARED_INTERMEDIATE_DIR)', '<!@(git describe --tags --always --abbrev=0)', '<!@(git rev-parse HEAD)'],
+          'action': ['<@(_inputs)', '<(SHARED_INTERMEDIATE_DIR)'],
```
Comment (jfirebaugh) [inline gyp/version.gypi]: What's the effect of this change?

Comment (jfirebaugh) [inline gyp/version.gypi]: Oh, I see, tag/rev parsing was merged into `build-version.py` itself.

**Comment by lucaswoj** [comment]
This is ready to go from my perspective but we're waiting until after the next iOS release to merge this because of the geojsonvt version bump.

File: configure (hunk 1 above)
Comment (mikemorris) [inline configure]: Maybe check for local mason as a git repo (submodule) if there's an earlier snippet somewhere initializing the mason submodule?
```
if [ -d "`pwd`/.mason" && -s "`pwd`/.mason/.git" ]; then
```

Comment (lucaswoj) [inline configure]: @mikemorris by "this" I was mainly referring to `$(dirname $(readlink $(which mason)))`

**Comment by mikemorris** [comment]
With iOS and Android released, this should be :checkered_flag: to merge @lucaswoj!

**Comment by incanus** [comment]
Correction: Android is release-branched but not yet out; however master is clear to merge as long as you aren’t anticipating any critical fixes very soon there that build upon this merge and that also apply to Android (which seems doubtful).

**Comment by lucaswoj** [comment]
Squashed into one commit, will merge when Travis gives the :white_check_mark:

**Comment by bleege** [comment]
This broke Android builds.  Running `make android` now results in...

```
deps/run_gyp geojsonvt.gyp -Iconfig.gypi --depth=. -Goutput_dir=. --generator-output=./build -f make
make -C build install
  CXX(target) Release/obj.target/geojsonvt/src/geojsonvt.o
  CXX(target) Release/obj.target/geojsonvt/src/geojsonvt_clip.o
  CXX(target) Release/obj.target/geojsonvt/src/geojsonvt_convert.o
  CXX(target) Release/obj.target/geojsonvt/src/geojsonvt_simplify.o
  CXX(target) Release/obj.target/geojsonvt/src/geojsonvt_tile.o
  LIBTOOL-STATIC Release/libgeojsonvt.a
error: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/libtool: file: Release/obj.target/geojsonvt/src/geojsonvt.o is not an object file (not allowed in a library)
error: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/libtool: file: Release/obj.target/geojsonvt/src/geojsonvt_clip.o is not an object file (not allowed in a library)
error: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/libtool: file: Release/obj.target/geojsonvt/src/geojsonvt_convert.o is not an object file (not allowed in a library)
error: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/libtool: file: Release/obj.target/geojsonvt/src/geojsonvt_simplify.o is not an object file (not allowed in a library)
error: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/libtool: file: Release/obj.target/geojsonvt/src/geojsonvt_tile.o is not an object file (not allowed in a library)
make[3]: *** [Release/libgeojsonvt.a] Error 1
make[2]: *** [install] Error 2
make[1]: *** [config/android-arm-v7.gypi] Error 2
make: *** [android-lib] Error 2
```

**Comment by mikemorris** [comment]
@bleege Did you cherry-pick this into the release branch and have problems? Travis was green on Android builds when we merged this...

**Comment by bleege** [comment]
@mikemorris Nope.  I just switched to https://github.com/mapbox/mapbox-gl-native/tree/2018-shared-sqlite first and saw that it failed.  I then nuked my entire local copy and re-cloned, ran the `git submodule init` and `git submodule update` commands, and then ran `make clean` and finally `make android`.  Same issue.

**Comment by bleege** [comment]
To help test, I did a partial revert in the https://github.com/mapbox/mapbox-gl-native/tree/2018-shared-sqlite branch to get things compile so that I can test #2018.  I'll wait for the SF folks to come on line later today before merging this into `master` as they'll probably have a way to fix things properly using the new way.

I should also clarify, I :heart: the direction of this to get away from git submodules and use Mason to roll dependencies.  I'm constantly stomping on git submodules so am looking forward to getting this working.

**Pull request merged**